sdk/synapse/Azure.Analytics.Synapse.Spark/src/Customization/SparkBatchOperation.cs

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Analytics.Synapse.Spark.Models;
+
+namespace Azure.Analytics.Synapse.Spark
+{
+    /// <summary>
+    /// A long-running operation for:
+    /// <see cref="SparkBatchClient.StartCreateSparkBatchJobAsync(SparkBatchJobOptions, bool?, CancellationToken)"/>,
+    /// <see cref="SparkBatchClient.StartCreateSparkBatchJob(SparkBatchJobOptions, bool?, CancellationToken)"/>,
+    /// <see cref="SparkBatchClient.StartGetSparkBatchJobAsync(int, bool?, CancellationToken)"/>,
+    /// <see cref="SparkBatchClient.StartGetSparkBatchJob(int, bool?, CancellationToken)"/>
+    /// </summary>
+    public class SparkBatchOperation : Operation<SparkBatchJob>
+    {
+        private static readonly TimeSpan s_defaultPollingInterval = TimeSpan.FromSeconds(5);
+
+        private readonly ClientDiagnostics _diagnostics;
+
+        private readonly SparkBatchClient _client;
+        private readonly SparkBatchJob _value;
+        private Response<SparkBatchJob> _response;
+        private bool _completed;
+
+        private static List<string> SessionSubmissionFinalStates = new List<string>
+        {
+            "error",
+            "dead",
+            "success",
+            "killed"
+        };
+
+        internal SparkBatchOperation(SparkBatchClient client, ClientDiagnostics diagnostics, Response<SparkBatchJob> response)
+        {
+            _client = client;
+            _value = response.Value ?? throw new InvalidOperationException("The response does not contain a value.");
+            _response = response;
+            _completed = false;
```

Comment:
default of `_completed` is false... does this need to be set here?

Reply:
Done.